machine/v1/types_nutanixprovider.go

```diff
@@ -154,6 +164,156 @@ type NutanixResourceIdentifier struct {
+	// The valid values are Standard and Flash, and the default is Standard.
+	// +kubebuilder:default=Standard
+	DiskMode NutanixDiskMode `json:"diskMode"`
+
+	// storageContainer refers to the storage_container used by the VM disk.
+	// +optional
+	StorageContainer *NutanixStorageResourceIdentifier `json:"storageContainer"`
+}
+
+// NutanixDiskDeviceType is the VM disk device type.
+// +kubebuilder:validation:Enum=Disk;CDRom
+type NutanixDiskDeviceType string
+
+const (
+	// NutanixDiskDeviceTypeDisk represents the VM disk device type "Disk".
+	NutanixDiskDeviceTypeDisk NutanixDiskDeviceType = "Disk"
+
+	// NutanixDiskDeviceTypeCDROM represents the VM disk device type "CDRom".
+	NutanixDiskDeviceTypeCDROM NutanixDiskDeviceType = "CDRom"
+)
+
+// NutanixDiskAdapterType is an enumeration of different disk device adapter types.
+// +kubebuilder:validation:Enum:=SCSI;IDE;PCI;SATA;SPAPR
+type NutanixDiskAdapterType string
+
+const (
+	// NutanixDiskAdapterTypeSCSI represents the disk adapter type "SCSI".
+	NutanixDiskAdapterTypeSCSI NutanixDiskAdapterType = "SCSI"
+
+	// NutanixDiskAdapterTypeIDE represents the disk adapter type "IDE".
+	NutanixDiskAdapterTypeIDE NutanixDiskAdapterType = "IDE"
+
+	// NutanixDiskAdapterTypePCI represents the disk adapter type "PCI".
+	NutanixDiskAdapterTypePCI NutanixDiskAdapterType = "PCI"
+
+	// NutanixDiskAdapterTypeSATA represents the disk adapter type "SATA".
+	NutanixDiskAdapterTypeSATA NutanixDiskAdapterType = "SATA"
+
+	// NutanixDiskAdapterTypeSPAPR represents the disk adapter type "SPAPR".
+	NutanixDiskAdapterTypeSPAPR NutanixDiskAdapterType = "SPAPR"
+)
+
+// NutanixVMDiskDeviceProperties specifies the disk device properties.
+type NutanixVMDiskDeviceProperties struct {
+	// deviceType specifies the disk device type.
+	// The valid values are "Disk" and "CDRom", and the default is "Disk".
+	// +kubebuilder:default=Disk
+	// +kubebuilder:validation:Required
+	DeviceType NutanixDiskDeviceType `json:"deviceType"`
+
+	// adapterType is the adapter type of the disk address.
+	// If the deviceType is "Disk", the valid adapterType can be "SCSI", "IDE", "PCI", "SATA" or "SPAPR".
+	// If the deviceType is "CDRom", the valid adapterType can be "IDE" or "SATA".
+	// +kubebuilder:validation:Required
+	AdapterType NutanixDiskAdapterType `json:"adapterType,omitempty"`
+
+	// deviceIndex is the index of the disk address. The valid values are non-negative integers, with the default value 0.
+	// For a Machine VM, the deviceIndex for the disks with the same deviceType/adapterType combination should start from 0
+	// and increase consecutively afterwards.
+	// +kubebuilder:default=0
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Required
+	DeviceIndex int32 `json:"deviceIndex,omitempty"`
+}
+
+// NutanixDataDisk specifies the VM data disk configuration parameters.
+type NutanixVMDisk struct {
+	// diskSize is size (in Quantity format) of the disk attached to the VM
+	// The minimum diskSize is 1Gi bytes.
```

Comment:
Can you add godoc explaining that this is a quantity, and explain the valid units/formatting?

Reply:
Added the link https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Format for the Quantity format and example documentation.